src/docs/devices/M5Stack-Fire/index.md

```diff
@@ -0,0 +1,212 @@
+
+## MPU6886 and BMM150
+
+![MPU6886 BMM150](MPU6886_BMM150.webp "MPU6886 BMM150")
+
+| I2C Address | Function                                |
+|-------------|-----------------------------------------|
+| 0x68        | MPU6886 6-Axis MotionTracking Sensor |
+| 0x10        | BMM150 3-Axis Geomagnetic Sensor     |
+
+## IP5306 Power Management
+
+M5Stack Fire uses a customized version of the IP5306 Power Management circuit
+[Customized IP5306 Datasheet]("https://github.com/m5stack/M5-Schematic/blob/master/Core/IIC_IP5306_REG_V1.4.pdf")
+
+| I2C Address | Function                                |
+|-------------|-----------------------------------------|
+| 0x75        | IP5306 Power Management |
+
+### IP5306 Example Configuration
+
+```yml
+external_components:
+  - source:
+      type: git
+      url: https://github.com/ssieb/custom_components
+    components: [ ip5306 ]
+
+ip5306:
+  battery_level:
+    name: Battery Level
+  charger_connected:
+    id: connected
+  charge_full:
+    id: full
+```
+
+Using the [IP5306 ESPHome Component]("https://github.com/ssieb/custom_components/tree/master/components/ip5306") by [@ssieb]("https://github.com/ssieb")
+
+## Example Configuration
+
+```yml
+substitutions:
+  friendly_name: "M5StackFire"
+
+esphome:
+  name: m5stackfire
+  platform: ESP32
+  board: m5stack-fire
+
+wifi:
+  ssid: !secret wifi_ssid
+  password: !secret wifi_password
+  
+  ap:
+    ssid: ${friendly_name} Fallback Hotspot
+    password: !secret wifi_password
+
+captive_portal:
+
+logger:
+
+api:
+
+ota:
+
+# Power Management
+external_components:
+  - source:
+      type: git
+      url: https://github.com/ssieb/custom_components
+    components: [ ip5306 ]
+
+i2c:
+  sda: 21
+  scl: 22
+  scan: True
+
+ip5306:
+  battery_level:
+    name: ${friendly_name} Battery Percent
+    id: battery_percent
+  charger_connected:
+    name: ${friendly_name} Charger Connected
+    id: connected
+  charge_full:
+    name: ${friendly_name} Charge Full
+    id: full
+
+# Buttons
+binary_sensor:
+  - platform: gpio
+    id: M5_BtnA
+    pin:
+      number: 39
+      inverted: true
+  - platform: gpio
+    id: M5_BtnB
+    pin:
+      number: 38
+      inverted: true
+  - platform: gpio
+    id: M5_BtnC
+    pin:
+      number: 37
+      inverted: true
+
+# Internal LEDS
+output:
+  - platform: ledc # backlight output
+    pin: 32
+    id: gpio_32_backlight_pwm
+
+light:
+  - platform: monochromatic # backlight
+    output: gpio_32_backlight_pwm
+    name: ${friendly_name} Backlight
+    id: backlight
+    restore_mode: ALWAYS_ON
+  - platform: fastled_clockless # side light
+    chipset: SK6812
+    pin: GPIO15
+    num_leds: 10
+    rgb_order: GRB
+    id: side_light
+    name: ${friendly_name} Side Light
+    restore_mode: ALWAYS_OFF
+
+# Deep Sleep
+deep_sleep:
+  id: deep_sleep_
+  wakeup_pin: 39
+  wakeup_pin_mode: INVERT_WAKEUP
+
+# Microphone and Speaker
+i2s_audio:
+  i2s_lrclk_pin: GPIO13
+  i2s_bclk_pin: GPIO32
+  i2s_mclk_pin: GPIO0
```

Comment:
```suggestion
```
GPIO32 is the display LED. AFAIK, you can leave bclk and mclk out and it will be set to I2S_PIN_NO_CHANGE (as it is in M5Stack source code too)